### PR TITLE
Integrate Docker with the main repo

### PR DESCRIPTION
**Pull request by isiko**
Before, the Docker image was in another repository and you had to build and run the Image via rather long commands manually. Now, the Image is contained in the same Repository and you can build and run it quickly via `docker-compose`, which should make it easier to start developing.

The only adjustment users of the Image have to make is to add a `.env` file, containing the `DISPLAY` variable used for X-Server integration, i.e. displaying GUIs outside of the container.

Also, I added a section to the `README.md` containing all information needed to run the ROS part of the project in Docker, so it is one of the first things you see when engaging with the project.

### CODE REVIEW

**Comment by RoldSI** [comment]
I would not integrate the Dockerfile into this repository.
This repository is only for robot operating code. Everything that is not part of the robot software should be stored separately. As the Dockerfile only provides a working environment for developers, it is not essential for robot operation. Hence, it should not be part of this repository.

This becomes relevant when code will be submitted for the FRE, and it will be deployed automatically to the physical robot. All unnecessary stuff should not be included because: (a) Clarity during development, (b) code review, and (c) performance (even if that is a minor concern).

**Comment by isiko** [comment]
I don't think the concerns you mentioned are valid, as they mostly aren't really touched by this PR or are affected more by other already existing Pars of the Repo. First of all, clarity during Development is probably more affected by having everything in one repo, although they really are developed separately, like the hoverboard and MCU code.
Secondly, this doesn't touch the aspect of Code review in any shape or form, as it doesn't affect any Code, and I also don't see why this should have a measurable impact on performance.

On the other hand, it improves the ease of development a lot, as the creation and removal of Docker containers is made a lot easier through the use of `docker-compose`. Also, this makes it easier for new developers to get started, as they mostly just run start developing, without the hassle of writing their own docker-compose file or memorizing long commands to start their container.

In conclusion, this PR has major positive Aspekts for Development, and minimal if not zero negative Impact on the Project.

**Comment by RoldSI** [comment]
> I also don't see why this should have a measurable impact on performance.
I already mentioned that this is a minor concern. But still, unnecessary stuff is unnecessary.

> First of all, clarity during Development is probably more affected by having everything in one repo, although they really are developed separately, like the hoverboard and MCU code.
This does only highlight: Separe things should be separate. And the development environment and the robot code are separate things.
But please make further suggestions for changes to the repo. But having the MCU code integrated helps as the robot will automatically flash connected devices. When deploying code, the robot only needs one code base instead of multiple ones.
Anyway, this should be discussed in a different place.

> On the other hand, it improves the ease of development a lot, as the creation and removal of Docker containers is made a lot easier through the use of docker-compose.
I do not disagree at all with the suggested changes. A docker-compose file is great.

> Also, this makes it easier for new developers to get started, as they mostly just run start developing, without the hassle of writing their own docker-compose file or memorizing long commands to start their container.
Docker information is still available. Just in a different place. My points do not at all impact the above-mentioned.
Furthermore, using a different place is also helpful as not all collaborators use Docker. Currently, only 2 out of 5 collaborators do actively use it. Thus, it is more annoying and might cause unforeseeable problems for the majority of the development team.
Besides, the effort for the minority to use a separate source for files is very low.

> In conclusion, this PR has major positive Aspekts for Development
I agree that the suggested Docker changes are helpful. But the location would only be beneficial to a minority. But, besides others, it leads to an inconsistent project structure.

**Comment by isiko** [comment]
> I already mentioned that this is a minor concern. But still, unnecessary stuff is unnecessary.

That's why I also didn't really mention that although I still don't think that this a concern even worth bringing up

> And the development environment and the robot code are separate things.

To a certain degree I think that's correct, and I also think that the Architecture of the whole Project should be discussed somewhere else, but I also think that this stands in contrast to b0425177611a8c139927b6185831d135c816fcdb If stuff like docker, which are just 2 Files for building and running the code, isn't wanted, why add those two files for building the code?
 
> Furthermore, using a different place is also helpful as not all collaborators use Docker. Currently, only 2 out of 5 collaborators do actively use it.

Again, I agree that this is not the place to discuss the Architecture of the Project, but there is a lot of other code that isn't used by a lot of us, for example, again, the hoverboard and MCU code.

> A docker-compose file is great.

Great we agree 👍 Problem is, having to write most of it before being able to use it makes it a bit useless, at least in eyes.

> Thus, it is more annoying and might cause unforeseeable problems 

If you don't use it, I don't see where the added 2 (!) files affect you, so how should it be annoying? Also: How should this have an effect in *any* way? The only software using the files in that folder are git, which shouldn't do anything unforeseen with that file, and colon, which has very specific folders it uses.

**Comment by isiko** [comment]
But if you still really don't want to add them to this repo, would you be OK with adding them as a submodule?

**Comment by RoldSI** [comment]
First, sorry for adding my scripts to the main repo. I have already removed them.

Anyway, to conclude, we will not add Docker information to this repo.
But please open another request in the docker repository!
